src/remotecontrol.cpp

```diff
@@ -72,32 +72,45 @@ void RemoteControl::handle()
     lastResult = result;
 
     auto &effectManager = g_ptrSystem->EffectManager();
+    auto& deviceConfig = g_ptrSystem->DeviceConfig();
 
     if (IR_ON == result)
     {
         debugV("Turning ON via remote");
         effectManager.ClearRemoteColor();
         effectManager.SetInterval(0);
         effectManager.StartEffect();
-        g_ptrSystem->DeviceConfig().SetBrightness(BRIGHTNESS_MAX);
+        deviceConfig.SetBrightness(BRIGHTNESS_MAX);
         return;
     }
     else if (IR_OFF == result)
     {
-        auto& deviceConfig = g_ptrSystem->DeviceConfig();
         deviceConfig.SetBrightness((int)deviceConfig.GetBrightness() - BRIGHTNESS_STEP);
         return;
     }
     else if (IR_BPLUS == result)
     {
-        effectManager.ClearRemoteColor();
-        effectManager.NextEffect();
+        if (deviceConfig.ApplyGlobalColors())
+        {
+            effectManager.ClearRemoteColor();
+            effectManager.Update();
+        } else
+        {
+            effectManager.NextEffect();
+        }

```

Comment:
This makes the behavior of IR_BPLUS (and IR_BMINUS below) depend on whether global colors are currently applied or not. Is this what you intend to do?

Also, please:

- Don't call `effectManager.Update()` directly; it will be called soon enough by the main drawing loop in drawing.cpp, - which is what that function is intended for.
- Put `else` on a line of its own.

Reply:
Yes, that is the desired behavior. Before I touched it, it stopped showing the temporary effect and went to the next effect in the queue. This minor change makes it so that it clears the temporary effect so that a person returns to the state before pressing the color button. It's the most logical way I have come up with yet to arrive at that result.

Regarding the call to the update function, that is good to know. I didn't know exactly when it was called, but figured it would be better to explicitly trigger. If it is not needed it will be eliminated.